recipes/newrelic/infrastructure/ohi/memcached/debian.yml

```diff
@@ -81,49 +66,107 @@ install:
     setup:
       label: "Installing Memcached integration..."
       cmds:
+        - sudo apt-get install libmemcached-tools -y > /dev/null
         - |
           sudo mkdir -p "/etc/newrelic-infra/integrations.d"
         - |
           # Get latest definitions and skip any failure because of deprecation
           sudo apt-get -o Acquire::Check-Valid-Until=false update -yq
         - |
-          sudo apt-get install nri-memcached -y
+          sudo apt-get install nri-memcached -y > /dev/null
         - |
           if [ -f /etc/newrelic-infra/integrations.d/memcached-config.yml ]; then
             sudo rm /etc/newrelic-infra/integrations.d/memcached-config.yml;
           fi
         - |
-          if [ -z {{.NR_CLI_DB_USERNAME}} ] && [ -z {{.NR_CLI_DB_PASSWORD}} ]; then
-            sudo tee /etc/newrelic-infra/integrations.d/memcached-config.yml > /dev/null <<"EOT"
+          
+          # try get values from env
+          NR_CLI_DB_USERNAME={{.NR_CLI_DB_USERNAME}}
+          NR_CLI_DB_PASSWORD={{.NR_CLI_DB_PASSWORD}}
+          NR_CLI_HOST={{.NR_CLI_HOST}}
+          NR_CLI_PORT={{.NR_CLI_PORT}}
+
+          # Set default values for variables
+          NR_CLI_HOST=${NR_CLI_HOST:-localhost}
+          NR_CLI_PORT=${NR_CLI_PORT:-11211}
+          MAX_RETRIES=3
+          TRIES=0 
+          CAN_CONNECT=$(curl $NR_CLI_HOST:$NR_CLI_PORT 2>&1 1>/dev/null -s -S | awk -F'[()]' '{print $2}')
+
+          while [[ $TRIES -lt $MAX_RETRIES  && $CAN_CONNECT == "7" ]]; do
+            printf "\nPlease enter the port number for your Memcached instance: "
+            read -r NR_CLI_DB_PORT
+            NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-11211}
```

Comment:
@NRhzhao I found an issue while testing the user entering a port.
It looks like we're capturing the port in a different variable name, notice the extra _DB. I let you fix the names to have a consistent one.
Also when prompting for port here, we shouldn't default to the typical port.

Reply:
fixed